core-libs/setup/ssr/optimized-engine/get-loggable-ssr-optimization-options.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SsrOptimizationOptions } from './ssr-optimization-options';
+
+/**
+ * Helper functions that maps optimization options to primitive values.
+ * This is useful for logging and monitoring purposes.
+ *
+ * @param value optimization options that should be logged
+ * @returns options containing only primitive values that are easier to read by developers and monitoring tools
+ */
+export const getLoggableSsrOptimizationOptions = (
+  value: SsrOptimizationOptions
+) => {
+  const newValue: Record<string, any> = { ...value };
+  Object.keys(value).forEach((key) => {
+    if (typeof newValue[key] === 'object') {
```

Comment:
[MINOR]

To be future proof (possible adding of config properties that are nested plain objects), let's print constructors only of objects that are not just plain objects
```suggestion
    // check for class instance, but not a plain object
    if (typeof newValue[key] === 'object' && plainObject.constructor !== Object) {
```